components/org.wso2.carbon.identity.remotefetch.core/src/main/java/org/wso2/carbon/identity/remotefetch/core/impl/handlers/repository/GitRepositoryManager.java

```diff
@@ -222,4 +227,40 @@ private boolean isFileInSubDirectory(File baseDir, File path) {
         }
         return isFileInSubDirectory(baseDir, path.getParentFile());
     }
+
+    private String sanitize(String input) {
+
+        if (StringUtils.isBlank(input)) {
+            return input;
+        }
+
+        return input.replaceAll("(\\r|\\n|%0D|%0A|%0a|%0d)", "");
+    }
+
+    private  boolean isSubDir(File baseFile, File userFile) {
+
+        Path baseDirPath = Paths.get(baseFile.getAbsolutePath());
+        Path userPath = Paths.get(userFile.getPath());
+
+        if (!baseDirPath.isAbsolute()) {
+            throw new IllegalArgumentException("Base path must be absolute");
+        }
+
+        if (userPath.isAbsolute()) {
+            throw new IllegalArgumentException("User path must be relative");
+        }
+
+    // Join the two paths together, then normalize so that any ".." elements
+    // in the userPath can remove parts of baseDirPath.
+    // (e.g. "/foo/bar/baz" + "../attack" -> "/foo/bar/attack")
+    final Path resolvedPath = baseDirPath.resolve(userPath).normalize();
+
+    // Make sure the resulting path is still within the required directory.
+    // (In the example above, "/foo/bar/attack" is not.)
+    if (!resolvedPath.startsWith(baseDirPath)) {
+            throw new IllegalArgumentException("User path escapes the base path");
+        }
+
+    return true;
+    }
```

Comment:
Formatting issue.

Reply:
Thank you, This comment was fixed in
https://github.com/wso2-extensions/identity-fetch-remote/pull/40
https://github.com/wso2-extensions/identity-fetch-remote/pull/41
https://github.com/wso2-extensions/identity-fetch-remote/pull/42
https://github.com/wso2-extensions/identity-fetch-remote/pull/43
https://github.com/wso2-extensions/identity-fetch-remote/pull/44
https://github.com/wso2-extensions/identity-fetch-remote/pull/45.
I resolve this conversation.